algorithms/src/std_algorithms/impl/Kokkos_RandomAccessIterator.hpp

```diff
@@ -111,77 +118,87 @@ class RandomAccessIterator< ::Kokkos::View<DataType, Args...> > {
 
   KOKKOS_FUNCTION
   reference operator[](difference_type n) const {
-    return m_view(m_current_index + n);
+    if constexpr (is_always_contiguous)
```

Comment:
would it make sense and make things a bit more readable than having all these ifs in many places to create a separate specialization of the iterator for contiguous views?

Reply:
I thought about it but decided against it. There are 6 places where it's used out of 21 functions. I think the amount of duplication would be too much, unless we use inheritance.